code/modules/cooking/recipes/recipe.dm

```diff
@@ -276,29 +278,33 @@
 					holder.remove_reagent(R.type, rvol-R.volume)
 
 
-	if (results.len > 1)
+	if (length(results) > 1)
 		//If we're here, then holder is a buffer containing the total reagents for all the results.
 		//So now we redistribute it among them
 		var/total = holder.total_volume
 		for (var/i in results)
 			var/atom/a = i //optimisation
-			holder.trans_to(a, total / results.len)
+			holder.trans_to(a, total / length(results))
 
 	return results
 
 //When exact is false, extraneous ingredients are ignored
 //When exact is true, extraneous ingredients will fail the recipe
 //In both cases, the full complement of required inredients is still needed
-/proc/select_recipe(var/list/datum/recipe/available_recipes, var/obj/obj as obj, var/exact = 0)
-	var/list/datum/recipe/possible_recipes = list()
-	for (var/datum/recipe/recipe in available_recipes)
+/proc/select_recipe(var/obj/obj as obj, var/exact = 0, var/appliance = 0)
+	if(!appliance)
+		return null
```

Comment:
We should never allow a recipe with no appliance, I think? If this is true we should produce an error message here or runtime -- And 0 shouldn't be set as the default arg value, it can just be null.